src/scripts/main.js

```diff
@@ -1,3 +1,317 @@
+    const numA = parseFloat(numberA);
+    const numB = parseFloat(numberB);
+
+    if (order === 'ASC') {
+      if (!isNaN(numA) && !isNaN(numB)) {
+        return numA - numB;
+      }
+
+      return valueA.localeCompare(valueB);
+    } else {
+      if (!isNaN(numA) && !isNaN(numB)) {
+        return numB - numA;
+      }
+
+      return valueB.localeCompare(valueA);
+    }
+  };
+
+  document.querySelectorAll('th').forEach((header, index) => {
+    let sortOrder = null;
+
+    header.addEventListener('click', () => {
+      // const table = header.closest('table');
+      const tbody = header.closest('table').querySelector('tbody');
+      // const table = header.closest('table');
+      // const tbody = table.querySelector('tbody');
+      const rowsArr = Array.from(tbody.querySelectorAll('tr'));
+
+      if (sortOrder === 'ASC') {
+        sortOrder = 'DESC';
+      } else {
+        sortOrder = 'ASC';
+      }
+
+      rowsArr.sort(sortFilter(index, sortOrder));
+
+      tbody.append(...rowsArr);
+    });
+  });
+
+  document.querySelectorAll('tbody tr').forEach((row) => {
+    row.addEventListener('click', () => {
+      document.querySelectorAll('tbody tr').forEach((r) => {
+        r.classList.remove('active');
+        r.style.backgroundColor = '';
+        r.style.color = '';
+      });
+
+      row.classList.add('active');
+      row.style.backgroundColor = 'rgb(208, 208, 208)';
+      row.style.color = 'black';
+    });
+  });
+
+  const form = document.createElement('form');
+
+  form.classList.add('new-employee-form');
+
+  const nameLabel = document.createElement('label');
+  const nameInput = document.createElement('input');
+
+  nameLabel.textContent = 'Name:';
+  nameInput.type = 'text';
+  nameInput.name = 'name';
+  nameInput.dataset.qa = 'name';
+  nameInput.required = true;
+  nameInput.setAttribute('data-qa', 'name');
+  nameLabel.appendChild(nameInput);
+
+  const positionLabel = document.createElement('label');
+  const positionInput = document.createElement('input');
+
+  positionLabel.textContent = 'Position:';
+  positionInput.type = 'text';
+  positionInput.name = 'position';
+  positionInput.dataset.qa = 'position';
+  positionInput.required = true;
+  positionInput.setAttribute('data-qa', 'position');
+  positionLabel.appendChild(positionInput);
+
+  const ageLabel = document.createElement('label');
+  const ageInput = document.createElement('input');
+
+  ageLabel.textContent = 'Age:';
+  ageInput.type = 'number';
+  ageInput.name = 'age';
+  ageInput.dataset.qa = 'age';
+  ageInput.required = true;
+  ageInput.setAttribute('data-qa', 'age');
+  ageLabel.appendChild(ageInput);
+
+  const salaryLabel = document.createElement('label');
+  const salaryInput = document.createElement('input');
+
+  salaryLabel.textContent = 'Salary:';
+  salaryInput.type = 'number';
+  salaryInput.name = 'salary';
+  salaryInput.dataset.qa = 'salary';
+  salaryInput.required = true;
+  salaryInput.setAttribute('data-qa', 'salary');
```

Comment:
Redundant code: The `setAttribute` method is used to set the `data-qa` attribute, which is already set using the `dataset` property. Consider removing one of these to avoid redundancy.